### ISSUE TITLE
Copy of Herald's Horn

### ISSUE THREAD

**Issue by christianhent** [magefree/mage#10517]
If you copy Herald's Horn for example, with The Mycosynth Gardens, this will result in an error in your next turn.

```
java.lang.NullPointerException
Server version: 1.4.52-V4-beta4 (build: 2023-06-17 14:20)
Stack trace:
mage.cards.h.HeraldsHornEffect.apply(HeraldsHorn.java:83)
mage.abilities.AbilityImpl.resolveMode(AbilityImpl.java:199)
mage.abilities.AbilityImpl.resolve(AbilityImpl.java:183)
mage.abilities.TriggeredAbilityImpl.resolve(TriggeredAbilityImpl.java:152)
mage.game.stack.StackAbility.resolve(StackAbility.java:87)
mage.game.GameImpl.resolve(GameImpl.java:1690)
mage.game.GameImpl.playPriority(GameImpl.java:1615)
mage.game.turn.Step.priority(Step.java:61)
mage.game.turn.Phase.playStep(Phase.java:184)
mage.game.turn.Phase.play(Phase.java:89)
mage.game.turn.Turn.play(Turn.java:125)
mage.game.GameImpl.playTurn(GameImpl.java:1087)
mage.game.GameImpl.play(GameImpl.java:997)
mage.game.GameImpl.start(GameImpl.java:973)
mage.server.game.GameWorker.call(GameWorker.java:34)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:748)
```

**Comment by xenohedron** [comment]
Presumably an issue with getting the chosen creature type

**Comment by xenohedron** [comment]
[[Herald's Horn]]

```
            SubType subtype = ChooseCreatureTypeEffect.getChosenCreatureType(source.getSourceId(), game); 
             filter.add(subtype.getPredicate());
```

Due to becoming copy without entering battlefield to make new choice, no corresponding source id in the value lookup. Returns null and fails to get predicate.

Real fix: figure out how a copy can access the correct saved value. Must research best technique.

**Comment by Grath** [comment]
If you somehow control a Herald's Horn with no chosen creature type, no spells will cost less to cast, not even creature spells with no creature type. You'll be able to look at the top card of your library at the beginning of each of your upkeeps, but you can never put it into your hand this way, even if it's a creature card with no creature type.
(2017-08-25)

**Comment by xenohedron** [comment]
Yeah that makes sense. I was interpreting the choice as a copiable value, but I guess it's not a characteristic?

707.2. When copying an object, the copy acquires the copiable values of the original object's characteristics and, for an object on the stack, choices made when casting or activating it (mode, targets, the value of X, whether it was kicked, how it will affect multiple targets, and so on). The copiable values are the values derived from the text printed on the object (that text being name, mana cost, color indicator, card type, subtype, supertype, rules text, power, toughness, and/or loyalty), as modified by other copy effects, by its face-down status, and by "as . . . enters the battlefield" and "as . . . is turned face up" abilities that set power and toughness (and may also set additional characteristics). Other effects (including type-changing and text-changing effects), status, counters, and stickers are not copied. 

If it's not a copiable value, then the fix is straightforward, though should still double check other cards that choose types in any case.

**Comment by Grath** [comment]
109.3. An object’s characteristics are name, mana cost, color, color indicator, card type, subtype, supertype, rules text, abilities, power, toughness, loyalty, defense, hand modifier, and life modifier. Objects can have some or all of these characteristics. Any other information about an object isn’t a characteristic. For example, characteristics don’t include whether a permanent is tapped, a spell’s target, an object’s owner or controller, what an Aura enchants, and so on.

**Comment by xenohedron** [comment]
I confused myself with reading the reference to "as... enters the battlefield" abilities and totally misinterpreted it the first time. Thanks for clarifying.

**Comment by xenohedron** [comment]
I looked at other usages and didn't find any other missing null checks (although I'm not totally sure if the conditional mana builders e.g. Cavern of Souls work correctly with Thespian's Stage)
